Improvements as per discussions and test plan notes

cristian-recoseanu opened the pull request:
- improve readability and conformance language as per discussions on the previous call and test plan notes
- add static roles for managers

cristian-recoseanu committed:
Update Framework.md

- to improve conformance language
- add static roles for managers

cristian-recoseanu committed:
Update Blocks.md

To improve readability as per discussions on the previous call.

cristian-recoseanu committed:
Update Introduction.md

To include paragraphs about standard and vendor specific models.

jonathan-r-thorpe commented on docs/Framework.md:
```diff
@@ -114,8 +114,8 @@ Nullable types are signaled using the `?` marker at the end of the type name.
 
-NcObject is the base abstract control class for any control class in the control model. Any other control class MUST be derived directly or indirectly from this class.  
-Further explanations are provided in a dedicated [NcObject](NcObject.md) section.
+NcObject is the base abstract control class for any control class in the control model.  
```
Possible simplification/clarification
```suggestion
NcObject is the abstract base class for all classes in the control model.  
```

jonathan-r-thorpe commented on docs/Framework.md:
```diff
@@ -114,8 +114,8 @@ Nullable types are signaled using the `?` marker at the end of the type name.
-NcObject is the base abstract control class for any control class in the control model. Any other control class MUST be derived directly or indirectly from this class.  
-Further explanations are provided in a dedicated [NcObject](NcObject.md) section.
+NcObject is the base abstract control class for any control class in the control model.  
+Further explanations and normative references are provided in a dedicated [NcObject](NcObject.md) section.
```
Suggested simplification
```suggestion
Further explanations and normative references are provided in the [NcObject](NcObject.md) section.
```

jonathan-r-thorpe commented on docs/Framework.md:
```diff
@@ -211,11 +211,11 @@ Further explanations are provided in a dedicated [Blocks](Blocks.md) section.
 ### Workers
 
-Further explanations are provided in a dedicated [Workers](Workers.md) section.
+Further explanations and normative references are provided in a dedicated [Workers](Workers.md) section.
```
Suggested simplification
```suggestion
Further explanations and normative references are provided in the [Workers](Workers.md) section.
```

jonathan-r-thorpe commented on docs/Framework.md:
```diff
@@ -211,11 +211,11 @@ Further explanations are provided in a dedicated [Blocks](Blocks.md) section.
 #### NcWorker
 
-NcWorker is the base worker control class for any worker control class in the control model. Vendor specific workers MUST be directly or indirectly derived from this control class.
+NcWorker is the base worker control class for any worker control class in the control model.
```
Suggested simplification
```suggestion
NcWorker is the base class for any worker control class in the control model.
```

jonathan-r-thorpe commented on docs/Introduction.md:
```diff
@@ -1,11 +1,15 @@
 
 The NMOS Control framework is a collection of class and datatype models and supporting documentation.
-It is aimed to serve as the foundation for building devices, components and features in a standard and reusable fashion.
+It is aimed to serve as the foundation for building devices, components and control and monitoring features in a standard and reusable fashion.
```
Suggested simplification
```suggestion
It serves as the foundation for building devices, components and control and monitoring features in a standard and reusable fashion.
```

jonathan-r-thorpe commented on docs/Introduction.md:
```diff
@@ -1,11 +1,15 @@
-The aggregated collection of control class and datatype models is published in the [Framework](Framework.md).
+The aggregated collection of control class and datatype models is published in the [Framework](Framework.md) with specific functionality published separately as [Feature sets](Feature%20sets.md).
+
+Models of both classes and datatypes published either in the [Framework](Framework.md) or in one of the [Feature sets](Feature%20sets.md) are deemed as `standard` models and MUST have their names prefixed with `"Nc"`.
```
Suggested simplification
```suggestion
Models of both classes and datatypes published either in the [Framework](Framework.md) or in one of the [Feature sets](Feature%20sets.md) are `standard` models and MUST have their names prefixed with `"Nc"`.
```

garethsb replied:
Why is the word standard in backticks, rather than single or double quotes (standard but not really standard 😉)?

Can vendors publish feature sets into the register? If so do they have to be prefixed with `Nc` too or not?

cristian-recoseanu replied:
Vendors cannot publish to the register directly. They would need to "elevate" any proposed models at which point they would become standard models.

cristian-recoseanu committed:
Update docs/Framework.md

Co-authored-by: jonathan-r-thorpe <64410119+jonathan-r-thorpe@users.noreply.github.com>

cristian-recoseanu committed:
Update docs/Framework.md

Co-authored-by: jonathan-r-thorpe <64410119+jonathan-r-thorpe@users.noreply.github.com>

cristian-recoseanu committed:
Update docs/Framework.md

Co-authored-by: jonathan-r-thorpe <64410119+jonathan-r-thorpe@users.noreply.github.com>

cristian-recoseanu committed:
Update docs/Framework.md

Co-authored-by: jonathan-r-thorpe <64410119+jonathan-r-thorpe@users.noreply.github.com>

cristian-recoseanu committed:
Update docs/Introduction.md

Co-authored-by: jonathan-r-thorpe <64410119+jonathan-r-thorpe@users.noreply.github.com>

cristian-recoseanu committed:
Update docs/Introduction.md

Co-authored-by: jonathan-r-thorpe <64410119+jonathan-r-thorpe@users.noreply.github.com>

cristian-recoseanu committed:
Improve clarity

by changing the way standard and vendor-specific key words are highlighted and hyphenation.

cristian-recoseanu commented:
Reviewed and updated on the call on the 6th of June 2023.
